vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/AuthenticationRequest.kt

```diff
@@ -1,16 +1,62 @@
 package at.asitplus.wallet.lib.oidc
 
-import io.ktor.http.*
+import at.asitplus.KmmResult.Companion.wrap
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
 
-sealed class AuthenticationRequestParametersFrom<T>(val source: T, val parameters: AuthenticationRequestParameters) {
-    class JwsSigned(
-        jwsSigned: at.asitplus.crypto.datatypes.jws.JwsSigned,
-        parameters: AuthenticationRequestParameters
-    ) : AuthenticationRequestParametersFrom<at.asitplus.crypto.datatypes.jws.JwsSigned>(jwsSigned, parameters)
+/**
+ * Helper class for allowing serialization of AuthenticationRequestParametersFrom<*>
+ */
+// TODO: possibly replace this with AuthenticationRequestParametersFrom<*> once it's serializable? or the other way?
+@Serializable
+data class AuthenticationRequest(
+    @Serializable
+    val source: AuthenticationRequestSource,
+    val parameters: AuthenticationRequestParameters,
+) {
+    fun serialize() = jsonSerializer.encodeToString(this)
 
-    class Uri(url: Url, parameters: AuthenticationRequestParameters) :
-        AuthenticationRequestParametersFrom<Url>(url, parameters)
+    fun toAuthenticationRequestParametersFrom(): AuthenticationRequestParametersFrom<*> {
+        return when(source) {
```

Comment:
Please convert to expression body

Reply:
fixed in commit fe9f81e9d270cad965ea36f45399ea19a3e3c22a